frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/disk-modal.tsx

```diff
@@ -142,35 +143,27 @@ export const DiskModal = withHandlePromise((props: DiskModalProps) => {
     (combinedDiskSize && combinedDiskSize.unit) || BinaryUnit.Gi,
   );
 
+  const [advancedDrawerIsOpen, setAdvancedDrawerIsOpen] = React.useState(false);
+
   const [accessMode, setAccessMode] = React.useState<AccessMode>(
-    isEditing ? AccessMode.fromString(dataVolume.getAccessModes()) : accessModes[0],
+    (isEditing && (combinedDisk.getAccessModes() || [])[0]) || null,
   );
 
   const [volumeMode, setVolumeMode] = React.useState<VolumeMode>(
-    isEditing ? VolumeMode.fromString(dataVolume.getVolumeMode()) : volumeModes[0],
+    (isEditing && combinedDisk.getVolumeMode()) || null,
   );
 
-  const [advancedDrawerIsOpen, setAdvancedDrawerIsOpen] = React.useState(false);
-
   React.useEffect(() => {
-    if (
-      source.requiresAccessModes() &&
-      source.requiresVolumeMode() &&
-      !isEditing &&
-      storageClassName
-    ) {
-      const setAccessAndVolumeModes = async () => {
-        const configMap = await getStorageClassConfigMap({ k8sGet });
-        const defaultAccessModes = getDefaultSCAccessModes(configMap, storageClassName);
-        const defaultVolumeMode = getDefaultSCVolumeMode(configMap, storageClassName);
-
-        setAccessMode(defaultAccessModes[0]);
-        setVolumeMode(defaultVolumeMode);
-      };
-
-      setAccessAndVolumeModes();
+    if (!isEditing && isLoaded(_storageClassConfigMap)) {
+      if (source.requiresVolumeMode()) {
+        setVolumeMode(getDefaultSCVolumeMode(storageClassConfigMap));
+      }
+      if (source.requiresAccessModes()) {
+        setAccessMode(getDefaultSCAccessModes(storageClassConfigMap)[0]);
+      }
     }
-  }, [isEditing, source, storageClassName]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isLoaded(_storageClassConfigMap)]);
```

Comment:
modal disabled until it loads ant then it should set the defaults only once and only for a new storage